src/main/java/com/school/project/service/impl/UserServiceImpl.java

```diff
@@ -51,7 +45,13 @@ public void delete(Long id) {
     @Override
     public User getUserById(Long id) {
         Optional<User> byId = userRepository.findById(id);
-        if (!byId.isPresent()) throw new UserNotFoundException("User with id :" + id + "is not found");
+        if (!userRepository.existsById(id)) throw new UserNotFoundException("User with id :" + id + "is not found");
```

Comment:
bad idea, we are opening two transactions to data base. you have already database response, why do you need to do it again?